FrameworkSystem/private/standardLogging/gLogging.py

```diff
@@ -0,0 +1,335 @@
+    logging.getLogger('root').getChild('log') == root.log == log child of root
+    """
+
+    # gLogging chain
+    self._childrens = []
+    self._parent = father
+
+    # all the different backends
+    self._backendsDict = {'stdout': StdoutBackend(),
+                          'stderr': StderrBackend(),
+                          'file': FileBackend(),
+                          'server': RemoteBackend()}
+
+    # initialize display options with the ones of the gLogging parent
+    if self._parent is not None:
+      self._options = self._parent.getDisplayOptions()
+    else:
+      self._options = {'showHeaders': True, 'showThreads': False, 'Color': False, 'Path': False}
+
+    # dictionary of the option state, modified by the user or not
+    # this is to give to the options the same behaviour that the logging setLevel()
+    self._optionsmodified = {'showHeaders': False, 'showThreads': False}
+
+    self._backendsList = []
+
+    self._logger = logging.getLogger(fathername).getChild(name)
+
+  def showHeaders(self, yesno=True):
+    """
+    Depending on the value, display or not the prefix of the message.
+    :params yesno: boolean determining the behaviour of the display
+    """
+    self._setOption('showHeaders', yesno)
+
+  def showThreadIDs(self, yesno=True):
+    """
+    Depending on the value, display or not the thread ID.
+    :params yesno: boolean determining the behaviour of the display
+    """
+    self._setOption('showThreads', yesno)
+
+  def _setOption(self, optionName, value):
+    """
+    Depending on the value, modify the value of the option.
+    This option will not be modified anymore. 
+    The options of the childrens will be updated if they were not modified before by a developer
+    :params optionName: string representing the name of the option to modify
+    :params value: boolean to give to the option  
+    """
+    self._options[optionName] = value
+    self._optionsmodified[optionName] = True
+    self._setChildrensDisplayOptions(optionName, self._options)
+    self._updateFormat()
+
+  def registerBackends(self, desiredBackends, backendOptions=None):
+    """
+    Attach a list of backends to the gLogging object.
+    :params desiredBackends: a list of different names attaching to differents backends.
+                             example : ['stdout', 'file', 'server']
+    :params backendOptions: a dictionary of different backend options. 
+                            example: {'FileName': '/tmp/log.txt'}
+    """
+    for stringHandler in desiredBackends:
+      stringHandler = stringHandler.lower()
+      stringHandler = stringHandler.strip()
+
+      # check if the name is correct
+      if stringHandler in self._backendsDict:
+        backend = self._backendsDict[stringHandler]
+
+        # check if the backend is already in the list
+        if backend not in self._backendsList:
+          if backendOptions is not None:
+            backend.setParameters(backendOptions.copy())
```

Comment:
I added a little comment. 
It is to avoid that the backends modify the original "backendOptions" dictionary, it is just a safety